fl4health/strategies/flash.py

```diff
@@ -0,0 +1,175 @@
+        """
+
+        super().__init__(
+            fraction_fit=fraction_fit,
+            fraction_evaluate=fraction_evaluate,
+            min_fit_clients=min_fit_clients,
+            min_evaluate_clients=min_evaluate_clients,
+            min_available_clients=min_available_clients,
+            evaluate_fn=evaluate_fn,
+            on_fit_config_fn=on_fit_config_fn,
+            on_evaluate_config_fn=on_evaluate_config_fn,
+            accept_failures=accept_failures,
+            initial_parameters=initial_parameters,
+            fit_metrics_aggregation_fn=fit_metrics_aggregation_fn,
+            evaluate_metrics_aggregation_fn=evaluate_metrics_aggregation_fn,
+            weighted_aggregation=weighted_aggregation,
+            weighted_eval_losses=weighted_eval_losses,
+        )
+        self.current_weights = parameters_to_ndarrays(initial_parameters)
+        self.eta = eta
+        self.eta_l = eta_l
+        self.tau = tau
+        self.beta_1 = beta_1
+        self.beta_2 = beta_2
+        self.m_t: Optional[NDArrays] = None
+        self.v_t: Optional[NDArrays] = None
+        self.d_t: Optional[NDArrays] = None
+
+    def __repr__(self) -> str:
+        """Compute a string representation of the strategy."""
+        rep = f"Flash(accept_failures={self.accept_failures})"
+        return rep
+
+    def _update_parameters(self, delta_t: NDArrays) -> None:
+        """Update m_t, v_t, beta_3, and d_t per-element"""
+        assert self.v_t is not None and self.m_t is not None and self.d_t is not None
+
+        for i, (delta, m, v, d_prev) in enumerate(zip(delta_t, self.m_t, self.v_t, self.d_t)):
+            # Update m_t
+            self.m_t[i] = self.beta_1 * m + (1 - self.beta_1) * delta
+
+            # Update v_t
+            self.v_t[i] = self.beta_2 * v + (1 - self.beta_2) * np.square(delta)
+
+            # Compute beta_3
+            norm_v_prev = np.abs(v)
+            norm_diff = np.abs(np.square(delta) - self.v_t[i])
+            beta_3_matrix = norm_v_prev / (norm_diff + norm_v_prev)
+
+            # Update d_t
+            self.d_t[i] = beta_3_matrix * d_prev + (1 - beta_3_matrix) * (np.square(delta) - self.v_t[i])
+
+    def aggregate_fit(
+        self,
+        server_round: int,
+        results: List[Tuple[ClientProxy, FitRes]],
+        failures: List[Union[Tuple[ClientProxy, FitRes], BaseException]],
+    ) -> Tuple[Optional[Parameters], Dict[str, Scalar]]:
+        """Aggregate fit results using the Flash method."""
+        fedavg_parameters_aggregated, metrics_aggregated = super().aggregate_fit(
+            server_round=server_round, results=results, failures=failures
+        )
+        if fedavg_parameters_aggregated is None:
+            return None, {}
+
+        fedavg_weights_aggregate = parameters_to_ndarrays(fedavg_parameters_aggregated)
+
+        delta_t: NDArrays = [x - y for x, y in zip(fedavg_weights_aggregate, self.current_weights)]
+        for attr in ["m_t", "v_t", "d_t"]:
```

Comment:
Not a big deal at all but can these be initialized in the constructor instead? Or am i missing something?

Reply:
I think it's worth setting these values in the `__init__`. Let us know if we're missing a reason that's not possible though, Justin.